src/components/SearchableBar/index.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState } from 'react';
+import styles from './index.module.css';
+import searchIcon from '../../assets/search.svg';
+
+interface Link {
+  id: string;
+  url: string;
+  title: string;
+  description: string;
+  // 필요한 경우 다른 속성들도 추가할 수 있습니다.
+}
+
+const SearchableBar = ({
+  links,
+  onSearch,
+}: {
+  links: Link[];
+  onSearch: (filteredLinks: Link[]) => void;
+}) => {
+  const [searchTerm, setSearchTerm] = useState('');
```

Comment:
```suggestion
  const [searchTerm, setSearchTerm] = useState<string>('');
```
'' 은 다양한 타입이 될 수 있어서 되도록 <string> 붙여주시는 게 좋아요!